resources/user-agents/browsers/nokia-browser/nokia-browser.json

```diff
@@ -62,581 +46,340 @@
-          "device": "Nokia 3660",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "Nokia5500d/*",
-          "device": "Nokia 5500d",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "Nokia6260/*",
-          "device": "Nokia 6260",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "Nokia6600/*",
-          "device": "Nokia 6600",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "Nokia6620/*",
-          "device": "Nokia 6620",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "Nokia6630/*",
-          "device": "Nokia 6630",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "Nokia6670/*",
-          "device": "Nokia 6670",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "Nokia6680/*",
-          "device": "Nokia 6680",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "Nokia6682/*",
-          "device": "Nokia 6682",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "Nokia7610/*",
-          "device": "Nokia 7610",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "Nokia7650/*",
-          "device": "Nokia 7650",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "Nokia9210i/*",
-          "device": "Nokia 9210i",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "NokiaE61-1/*",
-          "device": "Nokia E61",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "NokiaN-Gage/*",
-          "device": "Nokia N-Gage",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "NokiaN-GageQD/*",
-          "device": "Nokia N-GageQD",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "NokiaN91-1/*",
-          "device": "Nokia N91",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "NokiaN95_*",
-          "device": "Nokia N95",
-          "platforms": [
-            "SymbianOS"
-          ]
-        },
-        {
-          "match": "Nokia6120c/*",
-          "device": "Nokia 6120c",
+          "devices": {
+            "NokiaN95": "Nokia N95"
+          },
           "platforms": [
             "SymbianOS"
           ]
         },
         {
-          "match": "X700/*",
-          "device": "Nokia X700",
+          "match": "Nokia*/*",
```

Comment:
I added this pattern here to catch the useragents that have a Nokia device that wasn't added here yet.  Because of this pattern I did have to add a `Nokia*` type pattern to a couple of other browsers (UC Browser and Openwave I believe).

This is because they have a pattern like `* UCBrowser` which won't get looked at if something else that doesn't have an asterisk that early matches.